decentraland/screens/helpers.js

```diff
@@ -55,6 +62,40 @@ export const getContracts = () => {
   return contracts;
 };
 
+// Note: the value should be a string
+const _storeData = async (key, value) => {
+  try {
+    await SecureStore.setItemAsync(key, value);
+  } catch (error) {
+    showError("Unable to save data on storage.");
+  }
+};
+
+const _retrieveData = async key => {
+  try {
+    const value = await SecureStore.getItemAsync(key);
+    return value;
+  } catch (error) {
+    showError("Unable to retrieve data from storage.");
+  }
+};
+
+export const recoverOrCreateAccount = async () => {
+  const recoveredPrivateKey = await _retrieveData(EPHEMERAL_ACCOUNT_PRIV_KEY);
+
+  if (recoveredPrivateKey == null) {
+    // Note: The timeout for account creation is about ~20 secs.
+    // See more: https://github.com/tasitlabs/tasit/issues/42
+    const account = Account.create();
+    const { privateKey: createdPrivateKey } = account;
+    await _storeData(EPHEMERAL_ACCOUNT_PRIV_KEY, createdPrivateKey);
+    return account;
+  }
+
+  const account = createFromPrivateKey(recoveredPrivateKey);
```

Comment:
I suppose we could also persist some things from the account object itself if we don't want to frequently re-derive it? I'll have to give that more thought.

Reply:
That won't gate merging this PR, though. I think it's out-of-scope for this PR.

Reply:
Maybe `my assets` list? Refs: https://github.com/tasitlabs/tasit/issues/210

Reply:
That would be interesting to persist too. But I mean even persisting details of the account object itself that can be generated from the private key but might be slightly faster just to have ready.

Do you remember where that old issue was where you timed how long the `createFromPrivateKey` function takes to run?

Reply:
The issue is it: https://github.com/tasitlabs/tasit/issues/42
And I timed the `Account.create()` function, it's taking 20seg (I've counted again today).
I'll do that for `createFromPrivateKey` too.

Reply:
Okay great - thanks!

Reply:
`createFromPrivateKey` runs instantly

Reply:
Oh ok awesome, that's good to know

Reply:
In that case, no need to persist the rest of the account object separately

Reply:
Resolving this